### ISSUE TITLE
Strange wagon pulling behavior

### ISSUE THREAD

**Issue by Lordician** [capnkirok/animaniamod#95]
### Expected Behavior
1. Not being able to pull multiple carts/wagons at once.
2. Not being able to link carts/wagons.
3. Combination of 1 and 2 not being able to be done, linking carts/wagons to others that already have one linked to them.
### Actual Behavior
You can do all of those things, which can look very glitchy and might not be intended.
Aside of that, if you stop pulling a cart with multiple carts behind that one, they will occasionally fly up in the air for a very short moment.

### Steps to Reproduce
- Spawn wagon/cart.
-Right click wagon/cart to pull it yourself.
- Spawn another cart/wagon.
-Shift-Right Click to ride the next cart/wagon.
You have now linked the one you were pulling and the one you are sitting in.
Rinse and repeat, even link multiple to one for extra glitchy movement and behavior.

### Version of Minecraft, version of Animania, Single Player or Server
Minecraft 1.12.2
Forge 1.12.2-14.23.4.2705
Animania 1.5.1.1
Singleplayer, but i bet multiplayer is even worse.

### Screenshots encouraged
Already posted one in the discord, but i will put a different one here.
![2018-08-23_00 44 36](https://user-images.githubusercontent.com/26574603/44494842-b9950f80-a66d-11e8-92a2-d990669735dc.png)
QUICK-EDIT:
Here you can see the covered wagon and a couple of carts briefly lifting themselves from the ground trying to follow the cart i linked them to.

**Comment by capnkirok** [comment]
Yeah, that's a bug. Thanks for the detailed description and examples.
